### PR TITLE
Update stack templates to use a historical feature

### PR DESCRIPTION
**Pull request by thecoop**
Use a historical feature to indicate stack templates are supported

### CODE REVIEW

File: x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackFeature.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+package org.elasticsearch.xpack.stack;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.features.FeatureSpecification;
+import org.elasticsearch.features.NodeFeature;
+
+import java.util.Map;
+
+public class StackFeature implements FeatureSpecification {
```
Comment (ldematte) [inline x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackFeature.java:16]: Nit: to be consistent with other FeatureSpecification implementation, this should be plural (`StackFeatures`)

**Review by ldematte — APPROVED** [review]
LGTM, one comment about naming

**Review by dakrone — APPROVED** [review]
This LGTM also, but should we call it something like `StackTemplateFeature` since `StackFeature` is so generic (I know it's called `StackPlugin`, but that is not a great name either)?

**Comment by thecoop** [comment]
@elasticmachine update branch

**Pull request merged**